gremlin-go/driver/connection.go

```diff
@@ -42,15 +42,15 @@ type connection struct {
 }
 
 type connectionSettings struct {
-	authInfo         			*AuthInfo
-	tlsConfig         			*tls.Config
-	keepAliveInterval 			time.Duration
-	writeDeadline     			time.Duration
-	connectionTimeout 			time.Duration
-	enableCompression 			bool
-	readBufferSize				int
-	writeBufferSize				int
-	enableUserAgentOnConnect		bool
+	authInfo                 AuthInfoProvider
```

Comment:
I do have some concerns with potentially breaking things for some users by changing this type from `*AuthInfo` to `AuthInfoProvider`. As the fields in AuthInfo were all public, there are likely users who are modifying these fields directly instead of using the interface.

It might not be a big deal as the current expected use case is users construct the AuthInfo struct, pass it into a new client, then never touch it again. I don't believe the AuthInfo is publicly accessible anywhere after it has been passed into the client.

I want to think about this a little bit more but I believe it is ok and won't impact any users.

Reply:
> There appears to be one test failing in the checks right now. The maven logs can be a bit much to search through, I like to search for `❌` in the logs to find the failing test quickly. It appears that `gremlin-go/driver/gorillaTransporter.go:68` can sometimes result in a nil pointer dereference.

Thank you for the clarification on how to run the tests locally.

The issue seems to be one of the tests setting `authInfo` to `nil`, which is unexpected (and isn't done anywhere else in the codebase as far as I can tell) so now `gorillaTransporter.getAuthInfo()` has a nil-check and returns `NoopAuthInfo` if `nil`. `NoopAuthInfo` is an idempotent way of specifying a zero-value for the `AuthInfoProvider` and can be used publicly as well.

Reply:
> I don't believe the AuthInfo is publicly accessible anywhere after it has been passed into the client.

Since `DriverRemoteConnection.settings` itself is unexported (as per Go export rules, it would have to be named `Settings`) I don't see a way of how one would go about changing any setting after initialization.